### PR TITLE
Conveyors now gotta go fast!

### PR DESCRIPTION
**Pull request by PJB3005**
- Adds a new machinery process at 6 ds interval, which conveyors use.
- Boosts conveyor capacity to 25.
- Boosts mining & sorting capacity to 100 atoms/tick.

Idea stolen from https://github.com/tgstation/-tg-station/pull/10556

### CODE REVIEW

File: code/controllers/Processes/fast_machinery.dm [hunk 1]
```diff
@@ -0,0 +1,57 @@
+//Faster version of the one in machinery.dm
+
+var/global/list/fast_machines = list()
+/datum/controller/process/fast_machinery
+	schedule_interval = 10 // every 1 second.
```
Comment (Iamgoofball) [inline code/controllers/Processes/fast_machinery.dm]: Drop it to 0.5 seconds, it works without lag if you're going at 25 moved per second.

Comment (MrStonedOne) [inline code/controllers/Processes/fast_machinery.dm]: fuck that, tweak it until it the objects move in tune with the animation.

**Comment by Iamgoofball** [comment]
also I'm currently making all the mining smelters/stackers/thingiemajiggers run at fast machine speed to keep up with the belts/satisfy my inner factorio sense when using belts

File: code/controllers/Processes/fast_machinery.dm [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+					fast_machines.Cut(i, i + 1)
+
+		else
+			if(M)
+				M.inMachineList = 0
+
+			if(!fast_machines.Remove(M))
+				fast_machines.Cut(i, i + 1)
+
+		if(!(i % 500))
```
Comment (ComicIronic) [inline code/controllers/Processes/fast_machinery.dm]: What, why?

Comment (PJB3005) [inline code/controllers/Processes/fast_machinery.dm]: fuck if I knew, this file is literally a copy pasta from the machinery process with a lower interval.

Comment (RemieRichards) [inline code/controllers/Processes/fast_machinery.dm]: in 499/500 cases ```i % 500``` is simply ```i```
But where i is 500, it becomes ```500 % 500``` which is 0.

this essentially means it'll sleep every 500 machines processed.
Kind of odd, but hey.

Comment (ComicIronic) [inline code/controllers/Processes/fast_machinery.dm]: I understood the code.
I don't understand why it's HERE.

Comment (RemieRichards) [inline code/controllers/Processes/fast_machinery.dm]: I just told you, it sleeps every 500 processes, it's clearly an archaic leftover attempt at keeping processing cheap.

**Comment by SkowronX** [comment]
Way past cool

**Comment by bustygoku** [comment]
I spent like 20 minutes on this branch making a stupid conveyor belt loop. I think the server could handle EVEN FASTER CONVEYORS but that's just me.

**Comment by Iamgoofball** [comment]
@bustygoku Eh, this is about the optimal speed for them, any faster is way too fast, any slower is way too slow.

**Comment by clusterfack** [comment]
We need to condense down some of these redundant scheduler processes in the future

**Comment by Iamgoofball** [comment]
@clusterfack 
I've been running tests, as long as conveyors are capped to 25, you can stick as much shit on a line as you want and it won't pitch a tantrum.

**Comment by clusterfack** [comment]
Well let's do it then I guess, I'd rather this than *shudder* bring back the ore redemption machine

**Comment by Iamgoofball** [comment]
i don't see a problem with ore redemption machine, it lets me trade my ores in for swag and pokeball ripoffs

OH OH BEFORE YOU MERGE THIS

MAKE SURE ALL THE FUKKEN MINING MACHINES RUN ON FAST MACHINES

File: code/modules/recycling/conveyor2.dm [hunk 3]
```diff
@@ -88,6 +90,10 @@
 /obj/machinery/conveyor/New(loc, newdir = null, building = 0)
 	. = ..(loc)
 
+	//Gotta go fast!
+	machines -= src
+	fast_machines += src
```
Comment (Iamgoofball) [inline code/modules/recycling/conveyor2.dm]: put ALL the mining machines on this otherwise your belts will go fast but the machines will be hella slow FNR

Comment (clusterfack) [inline code/modules/recycling/conveyor2.dm]: He made the mining machines do 100 at once

Comment (Iamgoofball) [inline code/modules/recycling/conveyor2.dm]: but do they process every 0.5 seconds?

**Pull request merged**